File: code/ch02/2.1.1.twoSum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    int n = nums.length;
    Arrays.sort(nums);
    int l = 0;
    int r = n - 1;
    while (l < r) {
      if (nums[l] + nums[r] < target) {
        l += 1;
      } else if (nums[l] + nums[r] > target) {
        r -= 1;
      } else {
        return new int[] {l, r};
      }
    }
    return new int[] {};
  }
}

File: code/ch02/2.1.1.twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const n = nums.length;
  nums.sort((a, b) => a - b);
  let l = 0;
  let r = n - 1;
  while (l < r) {
    if (nums[l] + nums[r] < target) {
      l += 1;
    } else if (nums[l] + nums[r] > target) {
      r -= 1;
    } else {
      return [l, r];
    }
  }
  return [];
};
